### PR TITLE
all tasks

### PR DESCRIPTION
**Pull request by andreyzh777**

### CODE REVIEW

File: src/Task12/task12.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+function findMondey(string $timeStart, string $timeEnd): array
+{
+    $time = new DateTime($timeStart);
+    $end = new DateTime($timeEnd);
+    $result = [];
+    $intervalDay = new DateInterval("P1D");
+    $intervalM = new DateInterval("P1M");
+    while ((int) $time->format('j') != 1) {
```
Comment (Jampire) [inline src/Task12/task12.php:10]: Use strong comparison.

File: src/Task12/task12.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        $time->add($intervalDay);
+    }
+    while ($end > $time) {
+        if ((int) $time->format('w') == 1) {
```
Comment (Jampire) [inline src/Task12/task12.php:14]: Use strong comparison.

File: src/Task12/task12.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            array_push($result, (string)$time->format("d.m.Y"));
```
Comment (Jampire) [inline src/Task12/task12.php:15]: Use `$result[] = (string)$time->format("d.m.Y")` to add single element to the list.

File: src/Task12/task12.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        }
+            $time->add($intervalM);
+    }
+
+    return $result;
+}
+/*
+$arr = findMondey('01-01-1990', '31-12-1999');
+echo count($arr) . PHP_EOL;
+foreach ($arr as $data) {
```
Comment (Jampire) [inline src/Task12/task12.php:26]: Use `implode()` to print array.

File: src/Task13/task13.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+function delArray(array $a, int $elem): array
```
Comment (Jampire) [inline src/Task13/task13.php:3]: Wrong function name. Are you trying to delete whole array?
Wrong second argument name. It's not an element, but position number.

File: src/Task14/task14.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<?php
```
Comment (Jampire) [inline src/Task14/task14.php:1]: OK, but can be done with one `array_flip()`.

File: src/Task15/task15.php [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+function findElements(array $a): array
+{
+    $tempA = [];
+    $kolElementA = count($a);
```
Comment (Jampire) [inline src/Task15/task15.php:6]: English names only.

File: src/Task15/task15.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        }
+    }
+
+    return $tempA;
+}
+
+/*
+$a = [1,2,3,6,7,8,10];
+var_dump(findElements($a));
+*/
```
Comment (Jampire) [inline src/Task15/task15.php:22]: Wrong output.
Input:
```
[3,6,7,8,10]
```
Expected:
```
[
    [1] => 4
    [2] => 5
    [6] => 9
]
```
Actual:
```
[
    [3] => 4
    [4] => 5
    [8] => 9
]
```

File: src/Task18/task18.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+function searchEl(array $a): mixed
+{
+    $countA = array_count_values($a);
+    arsort($countA);
+    $elementCount = reset($countA);
+    if ($elementCount < count($a) / 2) {
+        return false;
```
Comment (Jampire) [inline src/Task18/task18.php:9]: What will you do, if `false` is actual array value?

File: src/Task18/task18.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    $nameElement = key($countA);
+
+    return $nameElement;
+}
+
+/*
+$array = [1, 2, 3, 4, 5, 5, 5, 5, 5, 5, 6];
+$element = searchEl($array);
+if ($element !== false) {
```
Comment (Jampire) [inline src/Task18/task18.php:19]: Move this calculation into the function.

**Review by Jampire — CHANGES_REQUESTED** [review]

File: src/Task15/task15.php (hunk 8 above)
Comment (andreyzh777) [reply src/Task15/task15.php:22]: it is not entirely clear how the conclusion is formed
what should be the output if?
Input:
[3,7,9,10]

File: src/Task18/task18.php (hunk 9 above)
Comment (andreyzh777) [reply src/Task18/task18.php:9]: false or true kill work script.
Script working with array string, int, double.

File: src/Task18/MyFunction.php [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+{
+    $countA = array_count_values($a);
+    arsort($countA);
+    $elementCount = reset($countA);
+    if ($elementCount < count($a) / 2) {
+        return false;
+    }
+    $nameElement = key($countA);
+
+    return $nameElement ? $nameElement : 'false';
```
Comment (Jampire) [inline src/Task18/MyFunction.php:13]: And if `'false'` is actual majority element?

**Pull request merged**